### DISCUSSION TITLE
Mysql 8.0 버전에서는 쿼리 캐시를 사용할 수 없어?

### DISCUSSION THREAD

**Discussion by Soyflat** [Well-Founded-Confidence/RealMySQL#30] [Q&A]
MySQL 8.0 버전에서도 쿼리 캐시(Query Cache)를 사용할 수 있습니다. 그러나 MySQL 8.0에서는 쿼리 캐시의 동작 방식이 변경되었습니다. `이전 버전의 MySQL에서는 쿼리 캐시가 전체 쿼리 결과를 캐시하여 사용자가 동일한 쿼리를 다시 실행할 때 결과를 즉시 반환`했습니다. 하지만 MySQL 8.0에서는 쿼리 캐시가 사용되는 경우가 제한되고, 동작 방식도 변경되었습니다.

MySQL 8.0에서는 InnoDB 버퍼 풀과 관련하여 다음과 같은 새로운 캐시 메커니즘이 도입되었습니다.

InnoDB 버퍼 풀에서 사용하는 캐시 메커니즘
새로운 캐시 API 및 뒷면에서 사용되는 캐시 메커니즘
이전 버전의 MySQL에서는 MyISAM 엔진이 사용하는 쿼리 캐시와 InnoDB 버퍼 풀을 모두 사용할 수 있었지만, `MySQL 8.0에서는 InnoDB 버퍼 풀에서 사용하는 캐시 메커니즘만 사용할 수 있습니다. 이것은 InnoDB 버퍼 풀과 쿼리 캐시가 서로 경쟁하지 않도록 하기 위함입니다`.

또한, MySQL 8.0에서는 쿼리 캐시의 동작 방식이 변경되어 쿼리 결과를 전체 캐시하지 않고 일부 캐시합니다. 이는 캐시의 유효성을 검사하는 데 필요한 리소스를 줄이기 위한 것입니다. 따라서 MySQL 8.0에서는 이전 버전과는 다른 캐시 동작 방식을 사용하지만, 여전히 쿼리 캐시를 사용할 수 있습니다.

**Comment by menthamin** [comment]
사용할 수 있지만 동작 방식이 변경되었다.

DB는 성능보다는 안정성이 우선이다.
- 캐싱 보다는 안정성 정합성 측면에서 8.0 쿼리 캐시 제거(?, 제거가 안되었을 수도..)

SNS 좋아요 테이블
- 쿼리 캐싱 기능을 빼는 것을 권장
- 업데이트가 잦다
- Contents, Like 를 분리
